crd/apis/network/v1/gkenetworkparamset_types.go

```diff
@@ -64,14 +64,77 @@ type NetworkRanges struct {
 	CIDRBlocks []string `json:"cidrBlocks"`
 }
 
+// GKENetworkParamSetConditionType is the type for status conditions on
+// a GKENetworkParamSet. This type should be used with the
+// GKENetworkParamSetStatus.Conditions field.
+type GKENetworkParamSetConditionType string
+
+const (
+	// GKENetworkParamSetStatusReady is the condition type that holds
+	// if the GKENetworkParamSet object is validated
+	GKENetworkParamSetStatusReady GKENetworkParamSetConditionType = "Ready"
+)
+
+// GKENetworkParamSetConditionReason defines the set of reasons that explain why a
+// particular GKENetworkParamSet condition type has been raised.
+type GKENetworkParamSetConditionReason string
```

Comment:
If I understand correctly, those are reasons for the NetworkConditionStatusParamsReady in Network object?

Reply:
Correct. The cross validated reasons are owned by the params object so I figured it made more sense here. What do you think?

Reply:
This makes sense. Lets rename this so it reflect what it is, since those are reasons for the Network's ParamsReady condition not GNP's condition. Provide proper comment of this, the current one does not mentions anything about Network.